### PR TITLE
[21569] Support for Topic Keys (vulcanexus-jazzy)

### PR DESCRIPTION
**Pull request by Mario-DL**
This PR adds topic instances support for `vulcanexus-jazzy`. 

**IMPORTANT NOTE**
This PR must be merged after 
* #32 
* #34

### CODE REVIEW

File: rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/TypeSupport.hpp [hunk 1]
```diff
@@ -133,9 +141,14 @@ class TypeSupport : public eprosima::fastdds::dds::TopicDataType
     const rosidl_message_type_support_t * type_supports
   );
 
-  bool max_size_bound_ {false};
-  bool is_plain_ {false};
+  bool max_size_bound_{false};
+  bool is_plain_{false};
+  bool key_is_unbounded_;
+  mutable size_t key_max_serialized_size_;
+  mutable eprosima::fastdds::MD5 md5_;
```
Comment (MiguelCompany) [inline rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/TypeSupport.hpp:148]: Include the header for MD5

File: rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/TypeSupport.hpp [hunk 2]
```diff
@@ -60,25 +61,26 @@ class TypeSupport : public eprosima::fastdds::dds::TopicDataType
...
   bool compute_key(
-    const void * const data,
-    eprosima::fastdds::rtps::InstanceHandle_t & ihandle,
-    bool force_md5 = false) override
+    eprosima::fastdds::rtps::SerializedPayload_t & /* payload */,
+    eprosima::fastdds::rtps::InstanceHandle_t & /* ihandle */,
+    bool /* force_md5 */) override
   {
-    (void)data; (void)ihandle; (void)force_md5;
     return false;
```
Comment (MiguelCompany) [inline rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/TypeSupport.hpp:76]: Add a comment indicating why this method can always return false.

File: rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp [hunk 3]
```diff
@@ -62,6 +62,58 @@ void * TypeSupport::create_data()
...
+        std::lock_guard lock(this->mtx_);
+        ret =
+          this->get_key_hash_from_ros_message(ser_data->data, &ihandle, force_md5, ser_data->impl);
+        break;
+      }
+
+    case FASTDDS_SERIALIZED_DATA_TYPE_CDR_BUFFER:
+      {
+        // TODO(Mario-DL)
+        // We would need a get_key_hash_from_payload method
```
Comment (MiguelCompany) [inline rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp:92]: We now have this method. Perhaps we could call it.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**